lib/auth/integration/integrationv1/service.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integrationv1
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	integrationpb "github.com/gravitational/teleport/api/gen/proto/go/teleport/integration/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// ServiceConfig holds configuration options for
+// the Integration gRPC service.
+type ServiceConfig struct {
+	Authorizer authz.Authorizer
+	Cache      services.IntegrationsGetter
+	Backend    services.Integrations
+	Logger     *logrus.Entry
+}
+
+// Service implements the teleport.integration.v1.IntegrationService RPC service.
+type Service struct {
+	integrationpb.UnimplementedIntegrationServiceServer
+	authorizer authz.Authorizer
+	cache      services.IntegrationsGetter
+	backend    services.Integrations
+	logger     *logrus.Entry
+}
+
+// NewService returns a new Integrations gRPC service.
+func NewService(cfg *ServiceConfig) (*Service, error) {
+	switch {
+	case cfg.Cache == nil:
+		return nil, trace.BadParameter("cache is required")
+	case cfg.Backend == nil:
+		return nil, trace.BadParameter("backend is required")
+	case cfg.Authorizer == nil:
+		return nil, trace.BadParameter("authorizer is required")
+	case cfg.Logger == nil:
+		cfg.Logger = logrus.WithField(trace.Component, "integrations.service")
+	}
```

Comment:
Why not do `CheckAndSetDefaults()` method on the config as we usually do?

Reply:
Sorry, copy paste from another service
Fixed :+1: